__tests__/server.test.js

```diff
@@ -315,4 +316,36 @@ describe('GET /api/users', () => {
             expect(errorMsg).toBe('404: Not Found')
         })
     })
+})
+describe('GET /api/articles?topic', () => {
+    test('200: responds with array of articles with matching topic', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(( { body } ) => {
+            const catArticles = body.articles
+            expect(Array.isArray(catArticles)).toBe(true)
+            expect(catArticles.length).toBe(1)
+            catArticles.forEach((catArticle) => {
+                expect(catArticle).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String),
+                })
+            })
+        })
+    })
+    test('404: responds with an error when the topic does not exist', () => {
+        return request(app)
+        .get('/api/articles?topic=spaghetti')
+        .expect(404)
+        .then(( { body } ) => {
+            const errorMsg = body.msg
+            expect(errorMsg).toBe('404: Not Found')
+        })
+    })
```

Comment:
Good error test, there's one more test you can add. 

There's a topic that does exist but doesn't have any articles attached to it yet. In that case we should return a 200 and an empty array